hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsSnapshot.java

```diff
@@ -65,259 +77,229 @@ public static void initClass() throws Exception {
 
-    createVolBuckKey(testVolBucket, testKey);
+    res = ToolRunner.run(shell,
+        new String[]{"-createSnapshot", bucketPath, snapshotName});
+    // Asserts that create request fails since snapshot name provided twice
+    Assertions.assertEquals(1, res);
+  }
 
-    OzoneFsShell shell = new OzoneFsShell(clientConf);
-    try {
-      int res = ToolRunner.run(shell,
-              new String[]{"-createSnapshot", testVolBucket, snapshotName});
-      // Asserts that create request succeeded
-      assertEquals(0, res);
-
-      res = ToolRunner.run(shell,
-              new String[]{"-createSnapshot", testVolBucket, snapshotName});
-      // Asserts that create request fails since snapshot name provided twice
-      assertEquals(1, res);
-    } finally {
-      shell.close();
-    }
+  /**
+   * Create snapshot should succeed.
+   * 1st case: valid snapshot name
+   * 2nd case: snapshot name length is less than 64 chars
+   */
+  @ParameterizedTest
+  @ValueSource(strings = {"snap-1",
+      "snap75795657617173401188448010125899089001363595171500499231286"})
+  public void testCreateSnapshotSuccess(String snapshotName)
+      throws Exception {
+    int res = ToolRunner.run(shell,
+        new String[]{"-createSnapshot", bucketPath, snapshotName});
+    // Asserts that create request succeeded
+    Assertions.assertEquals(0, res);
+
+    SnapshotInfo snapshotInfo = ozoneManager
+        .getMetadataManager()
+        .getSnapshotInfoTable()
+        .get(SnapshotInfo.getTableKey(volume, bucket, snapshotName));
+
+    // Assert that snapshot exists in RocksDB.
+    // We can't use list or valid if snapshot directory exists because DB
+    // transaction might not be flushed by the time.
+    Assertions.assertNotNull(snapshotInfo);
   }
 
-  @Test
-  public void testCreateSnapshotInvalidName() throws Exception {
-    String volume = "vol-" + RandomStringUtils.randomNumeric(5);
-    String bucket = "buc-" + RandomStringUtils.randomNumeric(5);
-    String key = "key-" + RandomStringUtils.randomNumeric(5);
-    String snapshotName = "snapa?b";
+  /**
+   * Create snapshot should fail.
+   * 1st case: snapshot name contains invalid char
+   * 2nd case: snapshot name consists only of numbers
+   * 3rd case: bucket path is invalid
+   * 4th case: snapshot name length is more than 64 chars
+   */
+  @ParameterizedTest
+  @ValueSource(strings = {"snapa?b", "1234",
+      "snap156808943643007724443266605711479126926050896107709081166294"})
+  public void testCreateSnapshotFailure(String snapshotName)
+      throws Exception {
+
+    int res = ToolRunner.run(shell,
+        new String[]{"-createSnapshot", bucketPath, snapshotName});
+    // Asserts that create request failed
+    Assertions.assertEquals(1, res);
+  }
 
-    String testVolBucket = OM_KEY_PREFIX + volume + OM_KEY_PREFIX + bucket;
-    String testKey = testVolBucket + OM_KEY_PREFIX + key;
+  @Test
+  public void testCreateSnapshotParameterMissing() throws Exception {
```

Comment:
What if we pass empty string for URI? Would it be wrong bucket path or bucket path is missing.

Just thinking if all the failure scenarios can be group to single parameterized test.

Reply:
I can't group it with the rest because it expects `-1` as the result integer instead of `1`. It actually tests `createSnapshot` without any parameters, since a snapshot without a name can be created.

Reply:
Optional: you could pass expected response in parameters similar to `expectedMessage`.
e.g.

```
  public void testCreateSnapshotFailure(String description,
                                        String paramBucketPath,
                                        String snapshotName,
                                        String expectedMessage,
                                        int expectedShellResponse)
```

Reply:
I can fix that!

Reply:
Thanks @xBis7